bytecomp/matching.ml

```diff
@@ -353,17 +380,19 @@ let jumps_add i pss jumps = match pss with
       else (i,(get_mins le_ctx (pss@qss)))::rem in
     add jumps
 
-
-let rec jumps_union (env1:(int*ctx list)list) env2 = match env1,env2 with
-| [],_ -> env2
-| _,[] -> env1
-| ((i1,pss1) as x1::rem1), ((i2,pss2) as x2::rem2) ->
-    if i1=i2 then
-      (i1,get_mins le_ctx (pss1@pss2))::jumps_union rem1 rem2
-    else if i1 > i2 then
-      x1::jumps_union rem1 env2
-    else
-      x2::jumps_union env1 rem2
+type jumps_union = (int * ctx list) list
```

Comment:
This type declaration is wrong (the function does a `_union` but the datatype is not a union); it should be `jumps`, which is already defined above, and you could use this to annotate `env1`  if you wanted to.

Reply:
I've fixed this and actually made a new `Jumps` module which cleans this up rather more satisfactorily.  It turns out that the type can just be made abstract.